clients/da-vinci-client/src/main/java/com/linkedin/davinci/VersionBackend.java

```diff
@@ -193,14 +193,30 @@ synchronized void tryStartHeartbeat() {
     if (isReportingPushStatus() && heartbeat == null) {
       heartbeat = backend.getExecutor().scheduleAtFixedRate(() -> {
         try {
-          backend.getPushStatusStoreWriter().writeHeartbeat(version.getStoreName());
+          sendOutHeartbeat(backend, version);
         } catch (Throwable t) {
           LOGGER.error("Unable to send heartbeat for {}", this);
         }
       }, 0, heartbeatInterval, TimeUnit.SECONDS);
     }
   }
 
+  protected static void sendOutHeartbeat(DaVinciBackend backend, Version version) {
+    if (backend.hasCurrentVersionBootstrapping()) {
```

Comment:
Hmm, maybe I still get it wrong, but this to me still has the same issue towards II logic - because in the beginning, all recovery/bootstrap work will happen in forked process and main process ingestion task map may not even have that SIT. So in that case - it will still send out status and regular HB. Maybe we should sync offline today to make sure we are aligned?

Reply:
But overall, I like current approach better than the delete HB one, and maybe we need to add [controller] tag, as it also take some changes there..